Command/HealthCheckCommand.php

```diff
@@ -36,13 +43,34 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $checkName = $input->getArgument('checkName');
         $runner = $this->getContainer()->get('liip_monitor.runner');
-        $runner->addReporter(new ConsoleReporter($output));
+
+        if ($input->getOption('nagios')) {
+            $runner->addReporter(new RawConsoleReporter($output));
+        } else {
+            $runner->addReporter(new ConsoleReporter($output));
+        }
         $runner->useAdditionalReporters($input->getOption('reporter'));
 
         if (0 === count($runner->getChecks())) {
             $output->writeln('<error>No checks configured.</error>');
         }
 
-        return $runner->run($checkName)->getFailureCount() ? 1 : 0;
+        /** @var \ZendDiagnostics\Result\Collection $results */
+        $results = $runner->run($checkName);
+        if ($input->getOption('nagios')) {
+            if ($results->getUnknownCount()) {
+                $returnCode = 3;
+            } else if($results->getFailureCount()) {
+                $returnCode = 2;
+            } else if($results->getWarningCount()) {
+                $returnCode = 1;
+            } else {
+                $returnCode = 0; // We may have som skipped although
+            }
+        } else {
+            $returnCode = $results->getFailureCount() ? 1 : 0;
+        }
+
+        return $returnCode;
```

Comment:
I would just return immediately on all these places. its always good to return early and this way we do not need the else either